core/src/main/kotlin/model/Documentable.kt

```diff
@@ -378,6 +378,21 @@ fun Documentable.dfs(predicate: (Documentable) -> Boolean): Documentable? =
         this.children.asSequence().mapNotNull { it.dfs(predicate) }.firstOrNull()
     }
 
+operator fun Documentable.iterator(): Iterator<Documentable> {
+    return flatTree().iterator()
+}
+
+fun Documentable.flatTree(): List<Documentable> {
```

Comment:
@Kordyjan .
I am not sure if the team likes something like this as public API?

Reply:
I'm totally ok with `flatTree` extension. I would only prefer a slightly different name for it, as it is not returning tree. `.withDescendats()` is what I would choose instead.
On the other hand, I'm not a fan of implementing the `iterator` operator on documentables. This allows for code like `for (d in someDocumentable) {...}` which may be confusing and is not much shorter than `for (d in someDocumentable.withDescendants()) {...}`.
Also, returning this as a lazily evaluated sequence instead of a list may be beneficial in some cases. Something like that is done for [pages](https://github.com/Kotlin/dokka/blob/77c8777b7f66bddd374d68decd507547d356d602/core/src/main/kotlin/pages/PageNodes.kt#L160), but I think it will be slightly more performant if we implement that as a recursive method having `SequenceScope` as a receiver, instead of sequence nesting.

Reply:
"withDescendats" 👌 Very nice! Thank you ☺️ 

Regarding the "iterator" API: Very good argument. I removed it ✅ 

Now to the "sequence" API: I do not think this will be always faster. For smaller trees, I would bet that just creating that list is much faster. However I am uncertain which dimensions this will have. So I switched to use the Sequence API. ✅